cli/medperf/logging/utils.py

```diff
@@ -0,0 +1,274 @@
+import csv
+import getpass
+import grp
+import logging
+import os
+import platform
+import re
+import shutil
+import socket
+import subprocess
+import tarfile
+
+import docker
+import pkg_resources
+import psutil
+import yaml
+
+from medperf import config
+from medperf.exceptions import ExecutionError
+
+
+def get_system_information():
+    # Get basic system information
+    system_info = {
+        "Platform": platform.platform(),
+        "Hostname": socket.gethostname(),
+        "Processor": platform.processor(),
+        "System Version": platform.version(),
+        "Python Version": platform.python_version(),
+    }
+    return system_info
+
+
+def get_memory_usage():
+    # Get memory usage
+    memory = psutil.virtual_memory()
+    return {
+        "Total Memory": memory.total,
+        "Available Memory": memory.available,
+        "Used Memory": memory.used,
+        "Memory Usage Percentage": memory.percent,
+    }
+
+
+def get_disk_usage():
+    # Get disk usage
+    disk_usage = psutil.disk_usage("/")
+    return {
+        "Total Disk Space": disk_usage.total,
+        "Used Disk Space": disk_usage.used,
+        "Free Disk Space": disk_usage.free,
+        "Disk Usage Percentage": disk_usage.percent,
+    }
+
+
+def get_user_information():
+    # Get user information
+    username = getpass.getuser()
```

Comment:
https://docs.python.org/3/library/getpass.html#getpass.getuser
possible exceptions could be raised